paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -181,8 +181,9 @@ void innerTransDataLayoutFromMKLDNN(DataLayout in_layout, DataLayout out_layout,
 
   if (in_format != out_format) {
     void* in_data = GetDataFromTensor(in, in_type);
-    const std::string key =
-        platform::CreateKey(in_tz, in_format, out_format, in_type);
+    std::string key =
+        platform::CreateKey(*dev_ctx, in_tz, in_format, out_format, in_type);
+    key = platform::ExtendKeyWithThreadingInfoIfNeeded(*dev_ctx, key);
```

Comment:
This key is already extended with ThreadInfo in ` platform::ReorderMKLDNNHandler` class constructor.